addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -1071,6 +1072,25 @@ export class WysiwygAdapterComponent extends Wysiwyg {
             type: $editable.data('oe-type'),
         };
     }
+    /**
+     * @override
+     */
+    _updateEditorUI(e) {
+        super._updateEditorUI(...arguments);
+        if (this.lastElement) {
+            const textOptionEl = getTextOptionTarget(this.lastElement);
+            // Text options elements (text animations, highlights) could be
+            // removed at any moment by the `SnippetsMenu` code. Since we decide
+            // to show / hide the toolbar by checking if `lastElement` is
+            // editable or not (`isContentEditable`), the check won't be correct
+            // if the element is actually removed from the DOM (`!isConnected`)
+            // If the `lastElement` is a text option element, we simply replace
+            // it by the parent of its text option wrapper.
+            if (textOptionEl) {
+                this.lastElement = textOptionEl.parentElement;
+            }
+        }
+    }
```

Comment:
I don't understand what this is about: the explanation does not seem to explain why this was not needed before this PR

Reply:
See: https://github.com/odoo/odoo/pull/158352#discussion_r1539703282 for full answer.